lib/translators/TerrainTranslator.js

```diff
@@ -35,37 +35,136 @@ const TerrainTranslator = {
         /*
          * Map size data
          */
-        outBuffer.addInt(terrainJson.map.width);
-        outBuffer.addInt(terrainJson.map.height);
+        outBuffer.addInt(terrainJson.map.width + 1);
+        outBuffer.addInt(terrainJson.map.height + 1);
 
-        // Unsupported
-        // map.offset is not implemented yet.. so we hardcode
-        outBuffer.addByte(0x00);outBuffer.addByte(0x00);outBuffer.addByte(0x80);outBuffer.addByte(0xc5);
-        outBuffer.addByte(0x00);outBuffer.addByte(0x00);outBuffer.addByte(0x80);outBuffer.addByte(0xc5);
-        //outBuffer.addFloat(terrainJson.map.offset.x);
-        //outBuffer.addFloat(terrainJson.map.offset.y);
+        // map offset
+        outBuffer.addFloat(terrainJson.map.offset.x);
+        outBuffer.addFloat(terrainJson.map.offset.y);
 
 
         /*
          * Tile points
          */
-        for(let i = terrainJson.tiles.length - 1; i >= 0; i--) {
+        for(let i = 0; i < terrainJson.tiles.length; i++) {
```

Comment:
Did you change the order in which tiles are stored? If so we'll want to make sure this is accurately documented on the wiki.

The old way read the array of tiles from bottom to top, because the (0, 0) point in the `.json` file is the top-left, but in the `.w3e` file the (0, 0) point is in the bottom left.

Reply:
yes I did (this was my last "problem" with the conversion), but it just clicked and I made that change. I didn't think about swapping around the read function to format it this way. We could definitely do that, I guess we could push to the front of the array instead of the back when reading? Or just update the wiki to make sure it's documented correctly. Thoughts?

Reply:
I think having the origin in the top-right makes it easier to read the JSON, since the array of tiles can be visualized just as you'd see it in the World Editor. This does have the unfortunate effect of having to do some extra coordinate changing behind the scenes, just as the original code did with reading the tile rows backward.

Could you switch the bounds on the for-loop on both warToJson and jsonToWar to reverse the rows as described earlier? It's a little more annoying, but I think the abstraction pays off for the end-user when they're looking at their JSON file.